src/browserExt/contentTypeHandler.js

```diff
@@ -137,15 +138,14 @@ Zotero.ContentTypeHandler = {
 	},
 
 	_isImportableStyle: function (url, contentType) {
-		const URI = new URL(url);
 		// Offer to install CSL by Content-Type
 		if (Zotero.ContentTypeHandler.cslContentTypes.has(contentType)) {
 			return true;
 		}
 		// Offer to install CSL if URL path ends with .csl and host is allowed
-		else if (URI.pathname.match(/\.csl$/)) {
+		else if (/\.csl$/i.test(url)) {
 			let hosts = Zotero.Prefs.get('allowedCSLExtensionHosts');
-			if (Array.isArray(hosts) && hosts.includes(URI.hostname)) {
+			if (Array.isArray(hosts) && hosts.some(host => new RegExp(host).test(url))) {
```

Comment:
Using regular expressions to test URLs is more compatible, but some variable names may need to be renamed